src/misc/test/confirm.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { confirm } from "../confirm";
+import { publish } from "../../pubSub/pubSub";
+import { stdToGeneratorPublishedMessage as prefix } from "../helpers";
+
+describe("confirm test", () => {
+  it("should return true when confirmed", async () => {
+    setTimeout(() => {
```

Comment:
Why publishing before subscribing? Why do we need the setTimeout?

I'm worried this could fail under some circumstances.

Reply:
Because the `await` locks the execution and so we cannot do it after calling the `confirm` method.